src/main/java/it/gov/pagopa/payment/service/payment/common/CommonAuthServiceImpl.java

```diff
@@ -117,6 +122,15 @@ protected void checkAuth(String trxCode, TransactionInProgress trx){
 
     }
 
+    protected void checkTrxStatusToInvokePreAuth(String userId, SyncTrxStatus status, TransactionInProgress trx) {
+        if ((status.equals(SyncTrxStatus.CREATED) && userId != null) ||
+                (status.equals(SyncTrxStatus.IDENTIFIED) && trx.getReward() == null)){
+           commonPreAuthService.previewPayment(trx,trx.getChannel(),SyncTrxStatus.AUTHORIZATION_REQUESTED);
+        }else if(!LIST_OF_STATUS_TO_IGNORE.contains(status)) {
+            trx.setStatus(SyncTrxStatus.AUTHORIZATION_REQUESTED);
```

Comment:
you have to store the status inside the database before to invoke rewardCalculator